### PR TITLE
ReplayRouteLocationConverter: reset cache variables on new route has set

### PR DESCRIPTION
**Pull request by RingerJK**
## Description

ReplayRouteLocationConverter, reset cached values on new route has been set 

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Please describe the PR goals. Just the stuff needed to implement the fix / feature and a simple rationale. It could contain many check points if needed

### Implementation

Please include all the relevant things implemented and also rationale, clarifications / disclaimers etc. related to the approach used. It could be as self code companion comments

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/location/replay/ReplayRouteLocationConverter.kt [hunk 1]
```diff
@@ -114,7 +116,8 @@ internal class ReplayRouteLocationConverter(
                 legs[currentLeg]?.steps()?.let { steps ->
                     steps[currentStep].geometry()
                 }
-            } ?: "", 6 // Use a precision of 6 decimal places when encoding or decoding a polyline
+            }
+                ?: "", 6 // Use a precision of 6 decimal places when encoding or decoding a polyline
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/location/replay/ReplayRouteLocationConverter.kt]: NIT unnecessary change

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/location/replay/ReplayRouteLocationConverter.kt [hunk 2]
```diff
@@ -52,6 +51,9 @@ internal class ReplayRouteLocationConverter(
 
     override fun setRoute(route: DirectionsRoute) {
         this.route = route
+        currentLeg = 0
+        currentStep = 0
+        time = 0
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/location/replay/ReplayRouteLocationConverter.kt]: Is this also a problem in the legacy 🤔? How are we handling it in the legacy implementation? https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/location/replay/ReplayRouteLocationConverter.kt

Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/location/replay/ReplayRouteLocationConverter.kt]: no, it's in new approach only.
Legacy recreate converter each time on new route has been set, no options to `reset` route(take a look a [constructor](https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/location/replay/ReplayRouteLocationConverter.kt#L12))
In new approach no need to recreate converter, only reset cache-values when new route is set

Comment (Guardiola31337) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/location/replay/ReplayRouteLocationConverter.kt]: Thanks for confirming 🙇
